wallet/wallet.h

```diff
@@ -38,6 +39,14 @@ enum wallet_output_type {
 	htlc_recv = 4
 };
 
+/* A database backed shachain struct. The datastructure is
+ * writethrough, reads are performed from an in-memory version, all
+ * writes are passed through to the DB. */
+struct wallet_shachain {
+	u64 id;
+	struct shachain chain;
+};
+
```

Comment:
Why doesn't this belong in wallet? I'd like to keep the db as model agnostic as possible (with the exception of the migrations). There are a few things that I'll more later such as `db_serialize_signature` and `sqlite3_column_pubkey`. Which are not wallet-related.